### ISSUE TITLE
[BUG]: Flipped colour chart

### ISSUE THREAD

**Issue by andieich** [colour-science/colour-checker-detection#73]
I try to colour-correct underwater photos with your package and the X-Rite colour chart.   

Usually, the detection works very well, but sometimes, there is an issue when the chart is detected as flipped. I uploaded an example [here](https://github.com/andieich/test_colourchart/blob/main/test_colourchart.ipynb).   
The colour chart is correctly detected in the image, but it is (wrongly) detected as flipped (this is displayed as a warning). I think that in this case, the order of `SWATCHES` is reversed. Like that, it does not fit anymore to the colour chart in the photo. I assume that this is because the colours are very off underwater. Is there a way to optimise the detection of the orientation of the colour chart? For me, this is quite obvious with the cyan and orange values.

**Comment by KelSolaar** [comment]
Hi @andieich,

It seems like you did not finish your sentence.

Cheers,

Thomas

**Comment by andieich** [comment]
@KelSolaar Thanks for your hint. You're right, something went wrong. I now updated the comment.

**Comment by KelSolaar** [comment]
Thanks! There is indeed something odd with the flipping happening here. I will take a look as I'm currently working on this.

**Comment by KelSolaar** [comment]
BTW, would it be possible to get your image for testing? Cheers!

**Comment by andieich** [comment]
Thanks! And sure, the photo is in the GitHub repository, [here](https://github.com/andieich/test_colourchart/blob/main/photos/test_photo.jpg) is the link.

**Comment by mexicantexan** [comment]
Several other macduff color libraries do a quick check with:
```python
def check_colorchecker(values, expected_values=expected_colors):
    """Find deviation of colorchecker `values` from expected values."""
    diff = (values - expected_values).ravel(order='K')
    return sqrt(np.dot(diff, diff))
```
so if we do a check after we gather the patches then we could do a quick check with:
```python
# determine which orientation has lower error
orient_1_error = check_colorchecker(patch_values)
orient_2_error = check_colorchecker(patch_values[::-1, ::-1])
```
then by comparing the errors we can flip the patch values if needed

**Comment by KelSolaar** [comment]
Thanks, this is a great suggestion and better than the way I was doing it! Sometimes the obvious... :)

**Comment by KelSolaar** [comment]
That should be fixed in an upcoming release:

![image](https://github.com/colour-science/colour-checker-detection/assets/99779/02ba167d-e854-4ea9-9d13-7a3cbb435541)

**Comment by KelSolaar** [comment]
This has been released in 0.2.0!

**Comment by andieich** [comment]
Thanks! Yes, now it is working and the detection is also improved
